packages/onr-core/src/mock/Mocks.ts

```diff
@@ -0,0 +1,98 @@
+import { Serializer, Server } from 'miragejs';
+import { FactoryDefinition, ModelDefinition } from 'miragejs/-types';
+import { getDebugger } from '@onr/common';
+import { Mock } from './Mock';
+
+const debug = getDebugger('onr:core:mocks');
+
+type IMocks = {
+  urlPrefix: string;
+  namespace: string;
+};
+
+type IModels = Record<string, ModelDefinition>;
+type IFactories = Record<string, FactoryDefinition>;
+type ISerializers = Record<string, Serializer>;
+
+export class Mocks implements IMocks {
+  protected mocks: Mock[] = [];
+  public urlPrefix: string;
+  public namespace: string;
+
+  constructor({ urlPrefix, namespace }: IMocks) {
+    this.urlPrefix = urlPrefix;
+    this.namespace = namespace;
+  }
+
+  getAll() {
+    return this.mocks;
+  }
+
+  addMock(mock: Mock): Mocks {
+    this.mocks.push(mock);
+
+    debug(`added mock => ${mock.resource}`);
+
+    return this;
+  }
+
+  addMocks(mocks: Mock[]): Mocks {
+    mocks.forEach(mock => this.addMock(mock));
+
+    return this;
+  }
+
+  registerRoutes(server: Server) {
+    if (this.urlPrefix) {
+      server.urlPrefix = this.urlPrefix;
+    }
+
+    if (this.namespace) {
+      server.namespace = this.namespace;
+    }
+
+    this.mocks.forEach(mock => mock.registerRoutes(server));
+
+    server.passthrough(request => {
+      if (request.url.startsWith('/_next/')) {
+        return true;
+      }
+    });
+  }
+
+  getSerializers(): ISerializers {
+    return this.mocks.reduce<ISerializers>(
+      (serializers: ISerializers, mock: Mock): ISerializers => {
+        if (mock.model) {
+          serializers[mock.resource] = mock.serializer;
+        }
+
+        return serializers;
+      },
+      {}
+    );
+  }
+
+  getModels(): IModels {
+    return this.mocks.reduce<IModels>((models: IModels, mock: Mock): IModels => {
+      if (mock.model) {
+        models[mock.resource] = mock.model;
+      }
+
+      return models;
+    }, {});
+  }
+
+  getFactories(): IFactories {
+    return this.mocks.reduce<IFactories>((factories: IFactories, mock: Mock): IFactories => {
```

Comment:
To `reduce` method, I used to declare the type as follow
```ts
Array.reduce((prev, curr, index) => {}, {} as type)
```
TS can automatically gen the type for `prev` and `curr` when return prev correctly

Reply:
Thanks master typescript!